.evergreen/cargo-test.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+CARGO_OPTIONS=()
+TEST_OPTIONS=("-Z unstable-options" "--format json" "--report-time")
+FEATURE_FLAGS=()
+CARGO_RESULT=0
+
+use_single_thread() {
+    if [ "${SINGLE_THREAD}" = true ]; then
+        TEST_OPTIONS+=("--test-threads=1")
+    fi
```

Comment:
it looks like we were only doing this for some tests, but I think the original intention of this flag was to run all of our tests single-threaded when it's set. maybe we should just inline this in `cargo_test`?

Reply:
Done.